### ISSUE TITLE
No warnings when the return value type is wrong.

### ISSUE THREAD

**Issue by 515hikaru** [Microsoft/pyright#29]
Thanks for publishing the great extension! I'm very glad to check types quickly in VSCode.

But when I tried it, I found an unexpected behavior. Pyright do not warn below code have type missmatch.

```python
def add(a: int, b: int) -> int:
    return float(a + b)
```

and

```python
def add(a: int, b: int) -> str:
    return a + b
```

`mypy` warns same code.

I confirmed this behavior in version 1.0.4.

**Comment by erictraut** [comment]
The `float` class provides an `__int__` method, so pyright assumes it can be assigned to any `int` value. Likewise, `int` provides a `__str__` method, so it can be assigned to any `str` value.  It's interesting that mypy flags this case. Does it generate an error or just a warning?

**Comment by 515hikaru** [comment]
Yes, mypy generates an error: `error: Incompatible return value type (got "float", expected "int")`

I used Python 3.7.2 and mypy 0.670.

**Comment by erictraut** [comment]
I added a partial fix in 1.0.5. The analyzer now correctly reports the error condition in your first example. It doesn't yet handle type inference for expressions that involve binary operations like addition, so your second example still doesn't report an error.

**Comment by erictraut** [comment]
I just published 1.0.7 which has a full fix. Pyright now infers types of arithmetic and logical expressions, so the examples you provided above work as expected.

**Comment by 515hikaru** [comment]
Great! Thank you.
